contracts/root/WithdrawManager.sol

```diff
@@ -0,0 +1,484 @@
+import { WETH } from "../token/WETH.sol";
+import { DepositManager } from "./DepositManager.sol";
+import { IRootChain } from "./IRootChain.sol";
+
+
+contract WithdrawManager is DepositManager {
+  using Merkle for bytes32;
+  using RLP for bytes;
+  using RLP for RLP.RLPItem;
+  using RLP for RLP.Iterator;
+
+  // 0x2e1a7d4d = sha3('withdraw(uint256)')
+  bytes4 constant private WITHDRAW_SIGNATURE = 0x2e1a7d4d;
+  // 0xa9059cbb = keccak256('transfer(address,uint256)')
+  bytes4 constant private TRANSFER_SIGNATURE = 0xa9059cbb;
+  // keccak256('Withdraw(address,address,uint256)')
+  bytes32 constant private WITHDRAW_EVENT_SIGNATURE = 0x9b1bfa7fa9ee420a16e124f794c35ac9f90472acc99140eb2f6447c714cad8eb;
+
+  //
+  // Storage
+  //
+
+  // structure for plasma exit
+  struct PlasmaExit {
+    address owner;
+    address token;
+    uint256 amount;
+  }
+
+  // all plasma exits
+  mapping (uint256 => PlasmaExit) public exits;
+
+  // exit queue for each token
+  mapping (address => address) public exitsQueues;
+
+  // exit NFT contract
+  address public exitNFTContract;
+
+  //
+  // Events
+  //
+
+  event Withdraw(
+    address indexed user,
+    address indexed token,
+    uint256 amount
+  );
+
+  event ExitStarted(
+    address indexed exitor,
+    uint256 indexed utxoPos,
+    address token,
+    uint256 amount
+  );
+
+  //
+  // Public functions
+  //
+
+  /**
+  * @dev Returns information about an exit.
+  * @param _utxoPos Position of the UTXO in the chain.
+  * @return A tuple representing the active exit for the given UTXO.
+  */
+  function getExit(uint256 _utxoPos)
+    public
+    view
+    returns (address, uint256)
+  {
+    return (exits[_utxoPos].token, exits[_utxoPos].amount);
+  }
+
+  /**
+  * @dev Determines the next exit to be processed.
+  * @param _token Asset type to be exited.
+  * @return A tuple of the position and time when this exit can be processed.
+  */
+  function getNextExit(address _token)
+    public
+    view
+    returns (uint256, uint256)
+  {
+    return PriorityQueue(exitsQueues[_token]).getMin();
+  }
+
+  /**
+  * @dev Processes any exits that have completed the exit period.
+  */
+  function processExits(address _token) public {
+    uint256 exitableAt;
+    uint256 utxoPos;
+
+    // retrieve priority queue
+    PriorityQueue exitQueue = PriorityQueue(exitsQueues[_token]);
+
+    // Iterate while the queue is not empty.
+    while (exitQueue.currentSize() > 0) {
+      (exitableAt, utxoPos) = getNextExit(_token);
+
+      // Check if this exit has finished its challenge period.
+      if (exitableAt > block.timestamp) {
+        return;
+      }
+
+      // get withdraw block
+      PlasmaExit memory currentExit = exits[utxoPos];
+
+      // process if NFT exists
+      // If an exit was successfully challenged, owner would be address(0).
+      address exitOwner = ExitNFT(exitNFTContract).ownerOf(utxoPos);
+      if (exitOwner != address(0)) {
+        if (_token == wethToken) {
```

Comment:
This condition scope had better to be separated function because possibly called from `withdrawal()` function.

Reply:
I didn't get what you saying.

Reply:
never mind, question solved by other answer.